src/vision/QAK/reading_data.py

```diff
@@ -0,0 +1,250 @@
+    pcd = o3d.geometry.PointCloud.create_from_rgbd_image(rgbd_image, intrinsic)
+    pcd.transform(pose)  # Apply the 4x4 transformation matrix
+
+    vis.add_geometry(pcd)
+    vis.poll_events()
+    vis.update_renderer()
+    point = pose[:3, 3]
+    points.append(point)
+    if visualize:
+        points.append(point)
+        line_set.points = o3d.utility.Vector3dVector(points)
+        if len(points) > 1:
+            lines = [[j, j + 1] for j in range(len(points) - 1)]
+            line_set.lines = o3d.utility.Vector2iVector(lines)
+            colors = [[1, 0, 0] for _ in range(len(lines))]  # czerwony kolor dla linii trajektorii
+            line_set.colors = o3d.utility.Vector3dVector(colors)
+
+        # TODO: add axis to the visualization
+        # transformed_axis = axis.transform(pose)
+        # vis.remove_geometry(axis)
+        # axis = transformed_axis
+        # vis.add_geometry(axis)
+        vis.update_geometry(line_set)
+        vis.add_geometry(pcd)
+        vis.poll_events()
+        vis.update_renderer()
+
+    # vis.clear_geometries()  # This line is commented out to keep previous point clouds
+    return rgb, pcd, pose[:3, 3]
+
+
+def init_simulation3D(base_path, visualize=False, poses=None):
+    images, sparse_depths, poses = load_data(base_path)
+    intrinsics_path = base_path + "/K.txt"
+    if visualize:
+        vis = o3d.visualization.Visualizer()
+        vis.create_window()
+        points = []
+        line_set = o3d.geometry.LineSet()
+        vis.add_geometry(line_set)
+        axis = o3d.geometry.TriangleMesh.create_coordinate_frame(size=0.1)
+        pose = read_pose(poses[0])
+        axis.transform(pose)
+        vis.add_geometry(axis)
+
+    return images, sparse_depths, poses, intrinsics_path, vis, points, line_set, axis
+
+
+def simulation_loop_reading_data(base_path, visualize=False):
+    """
+    Function to simulate reading data from the camera in loop frame by frame .
+    Args:
+    base_path (str): The base path where the data is stored.
+    visualize (bool): Whether to visualize the point cloud.
+    Returns:
+    none
+    """
+    images, sparse_depths, poses, intrinsics_path, vis, points, line_set, axis = init_simulation3D(base_path,
+                                                                                                   visualize=visualize)
+
+    for image_path, depth_path, pose_path in zip(images, sparse_depths, poses):
+        rgb = read_image(image_path)
+        depth = read_depth(depth_path)
+        pose = read_pose(pose_path)
+        rgb, pcd, point = simulation_reading_one_time(rgb, depth, pose, vis, intrinsics_path, visualize, points,
+                                                      line_set, axis)
+        ### HERE WE CAN ADD SOME FUNCTIONALITY
+        # TODO: ALL THE REST
+        ###
+
+    if visualize:
+        vis.run()
+        vis.destroy_window()
```

Comment:
jak mergowałś z mojego brancha to to nie push'uj tych zmian z swojego bo będziemy mieli konflikty zrób sobie `git rm -r --cached <file>`